tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationOptions.cs

```diff
@@ -29,6 +29,11 @@ internal static void LogException(Exception exception, string message = null)
         {
             // ReSharper disable twice ExplicitCallerInfoArgument
             Log.Error(exception, message ?? exception?.Message);
+            if (exception is AppSec.BlockException)
+            {
+                throw exception;
+            }
+
```

Comment:
This doesn't seem right to me 🤔 I'm not _entirely_ sure how we need/want to handle this, but we would probably need to update the call target generation rather than putting this in here. Alos, we don't want to rethrow here as it would lose the stack

Reply:
It's true that the current calltarget instrumentation avoid throwing from our code. And any `throw` from BeginMethod or EndMethod will be handled by `LogException`. That said, this solution seems a little hacky to me. I kind of prefer to adapt the calltarget machinery to support this, something like this: 
```csharp
using System;
public class C
{
    public void M()
    {
        
        try
        {
        }
        catch (CallTargetBubbleUpException bEx)
        {
            // We prefer to loose the stacktrace to avoid
            // having CallTargetInvoker.Begin<....>()
            // or CallTargetInvoker.End<....>()
            // inside a customer stacktrace?
            throw bEx;
        }
        catch (Exception ex)
        {
            // LogException(...)
        }
        
    }
    
    public abstract class CallTargetBubbleUpException : Exception 
    {
    }
    
    public class AppSecBlockException : CallTargetBubbleUpException {
    }
}


```

Reply:
yes I've been trying to add an exception filter in the call target generation but to no avail so far💦 , I'll remove this for now and try in another PR withing the `method_rewriter.cpp`

Reply:
for now I deleted this solution  in [ef58dfb](https://github.com/DataDog/dd-trace-dotnet/pull/3442/commits/ef58dfbb0ad8fc4f3297dac3eca7eb91655b72ea) and will try Tony's suggestion in another PR, thanks